edb/ir/ast.py

```diff
@@ -405,11 +406,25 @@ class Command(Base):
     __abstract_node__ = True
 
 
+@dataclasses.dataclass(frozen=True)
```

Comment:
Why a dataclass instead of an AST node?

Reply:
1. I don't think this should be visited by `dispatch.compile` (in particular because TypeRef here is a copy)
2. It contains mix of AST types: `ir.ast.TypeRef` and `schema.types.Type`
3. I would make it `Tuple` as in "source_map" but I prefer data classes instead of tuples.

Other than that, I didn't give it much thought.